hierarchy/store.go

```diff
@@ -68,13 +68,21 @@ func (store *Store) BuildHierarchy(instanceID, codeListID, dimensionName string)
 
 	// some ancestries may have data. Create a map with ancestries without data only.
 	// note: for big datasets we expect len(ancestries) << len(nodesWithData), so we iterate ancestries for efficiency
-	nodesWithoutData := map[string]struct{}{}
-	for id := range ancestryIDs {
+	nodesWithoutData := map[string]string{}
+	for id, code := range ancestryIDs {
 		if _, found := nodesWithData[id]; !found {
-			nodesWithoutData[id] = struct{}{}
+			nodesWithoutData[id] = code
 		}
 	}
 
+	// Create 'hasCode' edges if they don't exist
+	if err := store.CreateHasCodeEdges(ctx, 1, codeListID, nodesWithData); err != nil {
```

Comment:
what is `1` in this case? I can see it's used in multiple places so maybe consider creating a const for it with a name to describe what is represents.

Reply:
it's the attempt number, I will create a constant.